core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/MultipleChoiceEditorDirective.js

```diff
@@ -136,21 +136,30 @@ oppia.directive('multipleChoiceEditor', [
           $scope.saveChoice = function(index, newChoiceValue) {
             if (!newChoiceValue) {
               alertsService.addWarning('Cannot save an empty choice.');
-              return;
+              return {
```

Comment:
We haven't yet figured out a good way to do error handling across the code base. If we have an exception case, we should stop execution (so throwing makes sense), but we need to be careful to make sure:

1. The caller code is aware of the potential exception (though it should not catch it)
2. We somehow make sure the user is aware something bad just happened, since an exception is going to put the page in a bad state

If the case is not exceptional and can be recovered from, then returning an error code makes more sense. In that case, the callers need to also be aware of the returned value and should properly address both success and failure cases.